src/test.rs

```diff
@@ -281,21 +294,23 @@ impl<W: Worktree + Sync + Send + 'static> Manager<W> {
         }))
         .await?;
 
-        self.set_commits(commits)
+        self.set_commits(commits).await
     }
 
     // Inner non-async helper for set_revisions.
-    pub fn set_commits(&self, commits: impl IntoIterator<Item = Commit>) -> anyhow::Result<()> {
-        let mut job_cts = self.job_cts.lock();
+    pub async fn set_commits(&self, commits: impl IntoIterator<Item = Commit>) -> anyhow::Result<()> {
```

Comment:
The comment suggests it's important that this function isn't async, but now you made it async. Why might the author have wanted it to be non-async? Could we be creating a problem by making it async?

If it turns out to be an issue then perhaps we can avoid it by replacing `Commit` with some other kind of object representing a revision that needs testing, which optionally also includes the notes hash. Then we can pre-construct that object and pass it into this function, so that this function doesn't need to do the async hashing.